### PR TITLE
upgrade golang to 1.22.6

### PR DESCRIPTION
**Pull request by jeffreyc-splunk**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,10 @@
 - (Splunk) Add Azure Blob receiver ([#5200](https://github.com/signalfx/splunk-otel-collector/pull/5200))
 - (Splunk) Add Google Cloud PubSub receiver ([#5200](https://github.com/signalfx/splunk-otel-collector/pull/5200))
 
+### 💡 Enhancements 💡
+
+- (Splunk) Upgrade golang to 1.22 ([#5248](https://github.com/signalfx/splunk-otel-collector/pull/5248))
```
Comment (crobert-1) [inline CHANGELOG.md:16]: I believe this is a breaking change (that's what we've labeled it historically)

File: .github/dependabot.yml [hunk 2]
```diff
@@ -136,6 +136,10 @@ updates:
     directory: "/docker/httpd"
     schedule:
       interval: "weekly"
+  - package-ecosystem: "docker"
+    directory: "/docker/httpd-discovery"
+    schedule:
+      interval: "weekly"
```
Comment (crobert-1) [inline .github/dependabot.yml:142]: Out of ignorance, how is this related to the rest of the PR?

Comment (jeffreyc-splunk) [reply .github/dependabot.yml:142]: > Out of ignorance, how is this related to the rest of the PR?

Not sure, but the `gendependabot` job failed without it: https://github.com/signalfx/splunk-otel-collector/actions/runs/10426155220/job/28878539012

File: CHANGELOG.md (hunk 1 above)
Comment (jeffreyc-splunk) [reply CHANGELOG.md]: > I believe this is a breaking change (that's what we've labeled it historically)

Moved to breaking changes.

**Review by crobert-1 — APPROVED** [review]
